sdk/tables/Azure.Data.Tables/src/TablesExceptionParser.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Data.Tables
+{
+    internal static class TablesExceptionParser
+    {
+        /// <summary>
+        /// Parses a table service exception and populates the ErrorCode and Message, if possible.
+        /// </summary>
+        /// <param name="ex"></param>
+        /// <returns></returns>
+        public static RequestFailedException Parse(RequestFailedException ex)
+        {
+            if (TryParseOdataError(ex.Message, out string code))
+            {
+                var parsedException = new RequestFailedException(ex.Status, ex.Message, code, ex);
+                foreach (var key in ex.Data.Keys)
+                {
+                    parsedException.Data[key] = ex.Data[key];
+                }
+                return parsedException;
+            }
+            else
+            {
+                return ex;
+            }
+        }
+
+        /// <summary>
+        /// Parses an odata.error for its 'code'.
+        /// </summary>
+        private static bool TryParseOdataError(string error, out string errorCode)
+        {
+            // odaa.error payloads look like this:
+            // {"odata.error":{"code":"SomeErrorCode","message":{"lang":"en-US","value":"Some error message.\nRequestId:00000000-0000-0000-0000-000000000000\nTime:2021-01-01T12:00:00.0000000Z"}}}
+
+            var odataProp = "\"odata.error\":{".AsSpan();
```

Comment:
Why not use utf8json reader?

Reply:
Or JsonElement?

Reply:
I didn't initially because I still needed to parse the exception message string anyway to get to the json part for the default formatting. But the `ExtractFailureContent` method might fix that problem. Will investigate.